touchforms/formplayer/static/formplayer/script/xforgasm.js

```diff
@@ -119,6 +163,9 @@ function WebFormSession(params) {
     self.serverRequest = function (params, callback, blocking) {
         var that = this;
         var url = that.urls.xform;
+        if (params.action === 'submit-all' && this.NUM_PENDING_REQUESTS) {
+            self.taskQueue.addTask(params.action, self.serverRequest, arguments, self)
```

Comment:
Is `self` different from `this` and `that`? It's a little confusing to be using all three in this context when `this` would seem to work just fine.